src/main/java/com/upao/govench/govench/api/PostController.java

```diff
@@ -31,83 +42,166 @@ public class PostController {
     @Autowired
     private final CommunityService communityService;
 
-    @PostMapping("/create")
-    public ResponseEntity<String> createPost(@RequestBody PostRequestDTO postRequestDTO) {
+    @PostMapping("/community/{communityId}/posts/create")
+    public ResponseEntity<Map<String, String>> createPost(@PathVariable("communityId") int communityId,
+                                                          @RequestBody PostRequestDTO postRequestDTO) {
+
+        Integer userId = getAuthenticatedUserIdFromJWT();
+
+        if (userId == null) {
+            return new ResponseEntity<>(Map.of("message", "Acceso denegado: Usuario no autenticado"), HttpStatus.FORBIDDEN);
+        }
 
-        int userId = postRequestDTO.getAutor().getId();
         User author = userService.getUserbyId(userId);
 
         if (author == null) {
-            return new ResponseEntity<>("Usuario no encontrado", HttpStatus.NOT_FOUND);
+            return new ResponseEntity<>(Map.of("message", "Usuario no encontrado"), HttpStatus.NOT_FOUND);
```

Comment:
Porque se usa un Map.of?

Reply:
El Map.of en este caso se ha usado para devolver un mensaje o datos específicos (por ejemplo, body) en un formato clave-valor sin tener que crear un objeto adicional.